lib/vrf.c

```diff
@@ -377,6 +377,29 @@ vrf_bitmap_check (vrf_bitmap_t bmap, vrf_id_t vrf_id)
                      VRF_BITMAP_FLAG (offset)) ? 1 : 0;
 }
 
+static void
+vrf_autocomplete (vector comps, struct cmd_token *token)
+{
+  struct vrf *vrf = NULL;
+
+  RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name)
+    {
+      if (vrf->vrf_id != 0)
+        vector_set (comps, XSTRDUP (MTYPE_COMPLETION, vrf->name));
+    }
+}
+
+static const struct cmd_variable_handler vrf_var_handlers[] = {
+  {
+    .tokenname = "VRFNAME",
```

Comment:
this works -- just to note: you could have used `.varname = "vrf"` here (and not change all the DEFUNs to use VRFNAME).  It's documented in `cli.md`:

> Automatic assignment of variable names works by applying the following rules:
>
> - manual names always have priority
> - a "[no]" at the beginning receives "no" as varname on the "no" token
> - VARIABLE tokens whose text is not "WORD" or "NAME" receive a cleaned lowercase
>   version of the token text as varname, e.g. "ROUTE-MAP" becomes "route_map".
> - other variable tokens (i.e. everything except "fixed") receive the text of
>   the preceding fixed token as varname, if one can be found.  E.g.:
>   "ip route A.B.C.D/M INTERFACE" assigns "route" to the "A.B.C.D/M" token.

since all DEFUNs use "vrf NAME", and NAME is discarded for naming purposes, "vrf" is assigned as varname.

(with "VRFNAME", it will now assign "vrfname" as varname because unlike NAME it's not ignored)

This doesn't work for bgpd because it's `<view|vrf> NAME` so there is no name to use...